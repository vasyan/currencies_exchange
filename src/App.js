import React from 'react'
import { connect } from 'react-redux'
import Input from 'containers/input'
import Output from 'containers/output'
import { getRates } from 'actions/exchange'
import PhoneLayout from 'components/phoneLayout'
import styles from './styles.module.css'
import './index.css'

const CALL_API_TIMEOUT = 10000

class App extends React.Component {
  componentDidMount() {
    this.callApiInterval = setInterval(this.callApi, CALL_API_TIMEOUT)
    this.callApi()
  }

  componentWillUnmount() {
    clearInterval(this.callApiInterval)
    this.callApiInterval = null
  }

  callApi = () => {
    this.props.getRates()
  }

  render() {
    return (
      <PhoneLayout>
        <div className={styles.wrapper}>
          <Input />
          <Output />
        </div>
      </PhoneLayout>
    )
  }
}

function mapDispatchToProps() {
  return {
    getRates
  }
}

export default connect(
  null,
  mapDispatchToProps()
)(App)
