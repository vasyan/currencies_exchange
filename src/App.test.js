import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import App from './App'

function getStore({
  hasError = false,
  isOnFetch = true,
  isInitialFetched = false
} = {}) {
  return configureMockStore([thunk])({
    widgets: {
      exchange: {
        currencyFrom: 'FOO',
        currencyTo: 'BAR',
        amount: '42'
      }
    },
    collections: {
      rates: {
        USD: {
          FOO: 0.4242,
          BAR: 0.2424
        },
        hasError,
        isOnFetch,
        isInitialFetched
      }
    }
  })
}

describe('<App>', () => {
  it('should render loader on initial fetch', () => {
    expect(
      shallowToJson(shallow(<App store={getStore()} />).dive())
    ).toMatchSnapshot()
  })

  it('should render error message on problem with initial fetch', () => {
    expect(
      shallowToJson(
        shallow(
          <App store={getStore({ hasError: true, isOnFetch: false })} />
        ).dive()
      )
    ).toMatchSnapshot()
  })

  it('should fetch rates on mount', async () => {
    const store = getStore()
    shallow(<App store={store} />).dive()

    expect(store.getActions()).toContainEqual({ type: 'rates/GET_RATES_START' })
  })
})
