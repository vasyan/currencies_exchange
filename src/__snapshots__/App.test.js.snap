// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> should render 1`] = `
<_default>
  <div
    className="wrapper"
  >
    <Connect(Input) />
    <Connect(Output) />
  </div>
</_default>
`;
